[Ho Wei Haw] Duke Increments

wh-coder-2025 opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

wh-coder-2025 committed:
Add level-1 features: Greet, Echo, Exit

wh-coder-2025 committed:
Add level-1 features: Greet, Echo, Exit

wh-coder-2025 committed:
Add level-2 features: Add, List

wh-coder-2025 committed:
Add Level-3 features: Mark as Done

wh-coder-2025 committed:
Add Level-4 features: ToDos, Events, Deadlines

wh-coder-2025 committed:
Add Level-4 features: ToDos, Events, Deadlines

wh-coder-2025 committed:
Add Text-UI testing

wh-coder-2025 committed:
Merge pull request #1 from nus-cs2103-AY1920S1/master

Retrieving updates from original repository

wh-coder-2025 committed:
Add Level-5: Handles EmptyDescriptionException and
WrongInstructionException.

wh-coder-2025 committed:
Merge remote-tracking branch 'origin/master'

wh-coder-2025 committed:
Add Level-6 features: Delete task

wh-coder-2025 committed:
Edit Level-6 features: modified variable names in doneTask.

wh-coder-2025 committed:
Merge branches 'gradle' and 'master' of https://github.com/weihaw08/duke

wh-coder-2025 committed:
Add Level-7 Ability to save tasks in list

wh-coder-2025 committed:
Add Level-8 Ability to recognise dates and times

wh-coder-2025 committed:
Merge branch 'branch-Level-7'

wh-coder-2025 committed:
Merge branch 'branch-Level-8'

wh-coder-2025 committed:
Merged Level-7 and Level-8 to main branch.

wh-coder-2025 committed:
Edit Level-8: Fixed error in the conditionals.

wh-coder-2025 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/FormattedDate.java

wh-coder-2025 committed:
Add A-MoreOOP and A-Packages

wh-coder-2025 committed:
Minor style changes to Duke.java

wh-coder-2025 committed:
Minor edits made to duke.command.ByeCommand.java

wh-coder-2025 committed:
Add A-JUnit.
JUnit tests for classes in duke.tasks included.

wh-coder-2025 committed:
Add JUnit Testing for ToDo task.

wh-coder-2025 committed:
Corrected a typo in UI printTaskModification function.

wh-coder-2025 committed:
Add A-JavaDoc
JavaDoc comments made for every class in duke.command, duke.tasks and
duke.functionality.

wh-coder-2025 committed:
Add A-CodingStandard
Code adheres to the Google Java Coding Standard.

wh-coder-2025 committed:
Add Level-9 features: Able to find matches in keywords

wh-coder-2025 committed:
Merge branch 'branch-A-JavaDoc'

wh-coder-2025 committed:
Merge tag 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/ByeCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/DoneCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/functionality/Parser.java
#	src/main/java/duke/functionality/Storage.java
#	src/main/java/duke/functionality/TaskList.java
#	src/main/java/duke/functionality/Ui.java
#	src/main/java/duke/tasks/Event.java
#	src/main/java/duke/tasks/FormattedDate.java

wh-coder-2025 committed:
Merge branch 'branch-level-9'

# Conflicts:
#	src/main/java/duke/functionality/Parser.java
#	src/main/java/duke/tasks/FormattedDate.java

wh-coder-2025 committed:
Corrected the numbering of tasks when the find command is invoked.
Added functions that will check if the command is a find command in
Parser.

wh-coder-2025 committed:
Rearranged the test files for tasks into a new folder.

wh-coder-2025 committed:
Corrected some bugs in duke.AddCommand

wh-coder-2025 committed:
Refactor exception handling in Duke

wh-coder-2025 committed:
Add Level-10 features: Javafx GUI

wh-coder-2025 committed:
Merge branch 'A-Level-10'

fatclarence commented on src/main/java/duke/exception/InvalidTimeAndDateException.java:
```diff
@@ -3,8 +3,17 @@
 public class InvalidTimeAndDateException extends Exception {
     private static final String MESSAGE = "☹ OOPS!!! You are breaking the space-time continuum!";
```
I like the consideration of making this a "private static final" constant. Very good, something I shall adapt as well! Well done!

fatclarence commented on src/main/java/duke/tasks/FormattedDate.java:
```diff
@@ -51,4 +51,9 @@ private boolean isValidTime(String time) {
+
+    @Override
+    public int compareTo(FormattedDate anotherDate) {
+        return this.date.compareTo(anotherDate.date);
```
This is actually a really interesting way to do a check on whether the date is of the valid format that we want, that is "dd/mm/yyyy hhmm", however, may I ask what is the purpose of having a comparison instead of catching ParseExceptions and reporting them to the user? Still I really like this idea! But I would like to know your thought process behind this.

wh-coder-2025 replied:
I initially did that because it might be possible for the user to enter both the start and end dates in a valid format even though it doesn't make sense for the end date to be earlier than the start date.

However, I removed this function in my latest commit as I decided to make use of Java's LocalDateTime package which is able to throw an DateTimeParseException should the user enters an invalid date. As a result, I am still trying to figure out how to include this exception when the user enters an end date that is earlier than the start date for an event task.

wh-coder-2025 committed:
Complete A-CodeQuality

wh-coder-2025 committed:
Complete A-CodeQuality

wh-coder-2025 committed:
Merge branches 'branch-A-CodeQuality' and 'master' of https://github.com/weihaw08/duke into HEAD

wh-coder-2025 committed:
Add A-Assertions

wh-coder-2025 committed:
Merge pull request #8 from weihaw08/Branch-A-Assertions

Branch-A-Assertions

bakwxh commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+        dialogContainer.getChildren().addAll(userDialog, dukeDialog);
+        userInput.clear();
+        if (input.equals("bye")) {
+            Platform.exit();
```
It seems that the GUI will exit on its own when the user types "bye". Very good idea as it totally disables the user from making any additional input in the same session. I may consider doing this as well!

wh-coder-2025 committed:
Add C-Statistics

wh-coder-2025 committed:
Merge pull request #9 from weihaw08/branch-C-Statistics

Branch-C-Statistics

wh-coder-2025 committed:
Update test files

Test files are updated to adhere to changes in
duke.tasks.FormattedDateTime class.

wh-coder-2025 committed:
Merge pull request #10 from weihaw08/branch-C-Statistics

Update test files

wh-coder-2025 committed:
Edit function in Storage class

A minor edit is made in initialiseFile function.

wh-coder-2025 committed:
Edit DialogBox appearance

wh-coder-2025 committed:
Add representative screenshot of product

wh-coder-2025 committed:
Update README.md

User guide is now updated in README.md.

wh-coder-2025 committed:
Set theme jekyll-theme-architect

wh-coder-2025 committed:
Set theme jekyll-theme-cayman

wh-coder-2025 committed:
Change formatting in README.md

wh-coder-2025 committed:
Add Bye command guide in user guide

wh-coder-2025 committed:
Change package items and Duke display picture

TaskList class is moved to duke.tasks.

wh-coder-2025 committed:
Edit build.gradle file

wh-coder-2025 committed:
Make changes to behaviour of statistics

Number of tasks completed in the previous day no longer carry over to
the next day.
